opentelemetry-sdk/src/metrics/internal/mod.rs

```diff
@@ -48,40 +48,54 @@ impl Operation for Assign {
 ///
 /// This structure is parametrized by an `Operation` that indicates how
 /// updates to the underlying value trackers should be performed.
-pub(crate) struct ValueMap<T: Number<T>, O> {
+pub(crate) struct ValueMap<AU: AtomicallyUpdate<T>, T: Number<T>, O> {
```

Comment:
`ValueMap` struct takes three generic parameters now. There is scope to further refactor this. We could update `AtomicTracker` in a way that would allow us to get rid of `Operation` trait. I didn't want to move too many things around in this PR.